scripts/build/cpio_generator.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package build
+
+import (
+	"fmt"
+	gobuild "go/build"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func init() {
+	archiveGenerators["cpio"] = cpioGenerator{}
+}
+
+type cpioGenerator struct {
+}
+
+// This generator creates a cpio archive from the given list of files.
+// TODO: Replace this implementation with one which does not require sudo.
+// TODO: Preferably, it will share code with cmds/cpio.
+func (g cpioGenerator) generate(config Config, files []file) error {
```

Comment:
This is as good a time as any to fix this mess and get rid of the sh, find, and so on. 

if a sudo is needed then we need to make it so 
sudo build
gets us there, not embedded sh commands.